src/box/txn.c

```diff
@@ -447,6 +448,22 @@ txn_is_distributed(struct txn *txn)
 		txn->n_new_rows != txn->n_local_rows);
 }
 
+/** Wrapper function for calling space on_replace triggers. */
+static inline int
+txn_stmt_run_on_replace_triggers(struct txn_stmt *stmt)
+{
+	struct txn *txn = stmt->txn;
+	/*
+	 * Triggers see old_tuple and that tuple
+	 * must remain the same
+	 */
+	stmt->does_require_old_tuple = true;
+	txn->space_on_replace_triggers_depth++;
+	int rc = trigger_run(&stmt->space->on_replace, txn);
+	txn->space_on_replace_triggers_depth--;
+	return rc;
+}
+
```

Comment:
I don't see much point in this new function, cause it's called only once. Let's inline it. Same comment for `txn_check_on_replace_triggers_depth` if you manage to put it in `on_replace_dd_system_space`.

Reply:
fixed